woocommerce/woocommerce
# [Product Block Editor]: always show "loading" state when choosing suggested products

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





The changes introduced by this PR make that the app always show the "loading" state even when the data is not requested by the client (since it has already been stored)

Closes https://github.com/woocommerce/woocommerce/issues/43845



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use the new Product Editor
2. Go to the Linked Products section
3. Click on the `Choose products for me` again and again
4. Confirm the UI always hows the "loading" state
5. Additionally, you can take a look at the network tab. You should see only (the first) one request made by the client

https://github.com/woocommerce/woocommerce/assets/77539/fbba2310-b5dc-4c8f-8706-f2afc4595d5b



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
